src/AspNet.Security.OAuth.Bilibili/BilibiliAuthenticationHandler.cs

```diff
@@ -110,14 +109,14 @@ protected override async Task<AuthenticationTicket> CreateTicketAsync(
     {
         using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
         request.Headers.Add("access-token", tokens.AccessToken);
-        request.Headers.Add("x-bili-accesskeyid", Options.ClientId);
-        request.Headers.Add("x-bili-content-md5", "d41d8cd98f00b204e9800998ecf8427e"); // It's a GET request so there's no content, so we send the MD5 hash of an empty string
-        request.Headers.Add("x-bili-signature-method", "HMAC-SHA256");
-        request.Headers.Add("x-bili-signature-nonce", Base64Url.EncodeToString(RandomNumberGenerator.GetBytes(256 / 8)));
-        request.Headers.Add("x-bili-signature-version", "2.0");
-        request.Headers.Add("x-bili-timestamp", TimeProvider.GetUtcNow().ToUnixTimeSeconds().ToString(CultureInfo.InvariantCulture));
-
-        var signature = BuildSignatureString(request, Options.ClientSecret);
+
+        var xBiliHeaders = BuildXBiliHeaders();
+        foreach (var xbh in xBiliHeaders)
+        {
+            request.Headers.Add(xbh.Key, xbh.Value);
```

Comment:
Can this be deconstructed?
```suggestion
        foreach ((var name, var value) in xBiliHeaders)
        {
            request.Headers.Add(name, value);
```